### PR TITLE
impl(pubsub): add default pull lease manager class

### PR DESCRIPTION
**Pull request by alevenberg**
Part of https://github.com/googleapis/google-cloud-cpp/issues/13298

These are exact copies of the PullLeaseManager except with s/pull_lease_manager/default_pull_lease_manager and other variations.

In the next PR I will turn PullLeaseManager into an interface so the PullAckHandler can hold either a DefaultPullLeaseManager or a TracingPullLeaseManager{DefaultPullLeaseManager}

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13355)

### CODE REVIEW

File: google/cloud/pubsub/internal/default_pull_lease_manager.cc [hunk 1]
```diff
@@ -0,0 +1,134 @@
+// Copyright 2022 Google LLC
```
Comment (coryan) [inline google/cloud/pubsub/internal/default_pull_lease_manager.cc:1]: nit: copyright year

**Review by coryan — APPROVED** [review]
It may be clearer (when looking at the history of these files) if you *renamed* the files and classes in a single PR instead of copying them and then renaming things.

File: google/cloud/pubsub/internal/default_pull_lease_manager.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/default_pull_lease_manager.cc:1]: Ack, changed in the new PR

**Comment by alevenberg** [comment]
I didn't know how to save this PR, but created a new one. TIL: `git mv`.

Moved the files then cherry picked the commit from this PR. Also, updated the copyright year.

https://github.com/googleapis/google-cloud-cpp/pull/13365

**Pull request closed without merging**